frontend/src/scenes/funnels/funnelLogic.test.ts

```diff
@@ -399,15 +420,35 @@ describe('funnelLogic', () => {
                                 failure_count: 1,
                                 result_type: FunnelCorrelationResultsType.Properties,
                             },
+                            {
+                                event: { event: 'another property' },
+                                success_count: 1,
+                                failure_count: 1,
+                                result_type: FunnelCorrelationResultsType.Properties,
+                            },
                         ],
                     },
                 })
         })
 
         it('triggers request to correlation endpoint when property excluded', async () => {
-            await expectLogic(logic, () => logic.actions.excludeProperty('another property'))
+            await expectLogic(logic, () => {
+                logic.actions.setPropertyNames(logic.values.allProperties)
+                logic.actions.excludeProperty('another property')
+            })
+                .toMatchValues({
+                    propertyNames: ['some property', 'third property'],
+                    excludedPropertyNames: ['another property'],
+                    allProperties: ['some property', 'third property'],
+                })
+                .toDispatchActions(logic, ['loadPropertyCorrelations'])
```

Comment:
What's the need for these checks btw if we get toMatchValues evaluating to true? It seems like testing the implementation when we're already testing the interface.

Reply:
Hmmm, since there can be other listeners on an action, it's not as clear cut (to me) that values is a real interface. Sure, the values might be correct, but the unit of work is an action, and different actions can have the same results, so I want to ensure that the correct actions are triggered always, no matter the values I'm checking.

For example (contrived): Another logic listens to `loadPropertyCorrelations`, and the values here were populated from a pre-existing "cache" / duplicate action in code. This test would pass, even if the wrong action was triggered. But, this will mean the other logic which depends on it, in some other part of the codebase would fail.

Reply:
👍 

In these cases where we're specifically expecting an action to fire as its part of the public interface and contract that other logics may depend on yes makes sense. I wonder if we can define the public interface explicitly, maybe not worth it though.